### PR TITLE
Update build, deploy and flash with new nanoff capabilities

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Update build, deploy and flash with new nanoff capabilities

## Motivation and Context

- Update build, deploy and flash with new nanoff capabilities
- Remove call to deployer and replace with nanoff
- Adjust all paths and calling tool
- Sorting the device versions properly
- Fixing issues with board naming
- Adding Skylabs devices capability to flash

## How Has This Been Tested?

On Windows
Need to be tested on a Mac/Linux

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: src/dotnet.ts [hunk 1]
```diff
@@ -39,62 +39,41 @@ export class Dotnet {
     public static deploy(fileUri: string, serialPath: string, toolPath: String) {
         if (fileUri) {
             const outputDir = path.dirname(fileUri) + '/OutputDir/';
-            const cliBuildArguments = `/p:NanoFrameworkProjectSystemPath=${toolPath}/nanoFramework/v1.0/ /p:OutDir=${outputDir}`;
-            const cliDeployArguments = `${toolPath}/nanoFrameworkDeployer/nanoFrameworkDeployer.exe -v ${serialPath ? '-c '+ serialPath : ''} -d ${outputDir}`;
+            const cliBuildArgumentsLinux = `/p:NanoFrameworkProjectSystemPath=${toolPath}/nanoFramework/v1.0/ /p:OutDir=${outputDir}`;
+            const cliBuildArgumentsWindows = `/p:NanoFrameworkProjectSystemPath=`+ toolPath + `\\nanoFramework\\v1.0\\ /p:OutDir=${outputDir}`;
+            const cliDeployArgumentsLinux = `${toolPath}/nanoFirmwareFlasher/nanoff.dll --nanodevice --deploy --serialport  ${serialPath} --image ${outputDir}`;
```
Comment (josesimoes) [inline src/dotnet.ts:44]: Shouldn't all these calls be EXE even on Linux?

Comment (Ellerbach) [reply src/dotnet.ts:44]: it's dll that is called, not the exe

**Comment by Ellerbach** [comment]
@josesimoes any idea why the build fails on the serial USB? Any similar issue in the past?

**Comment by josesimoes** [comment]
@Ellerbach can't recall seeing this before... I've re-run just in case it was a transient problem.
If its not, neeeds to be investigated. Maybe there is a new version of the `serialport` thing...

File: src/dotnet.ts (hunk 1 above)
Comment (josesimoes) [reply src/dotnet.ts:44]: OK just wanted to be sure. (You know **nix is not my thing... 😅 )

**Comment by Ellerbach** [comment]
> Maybe there is a new version of the `serialport` thing...

I guess it's about this. So need some investigation. I can try by the end of the week. If you have extra time, feel free to look at it ;-)

**Comment by networkfusion** [comment]
@Ellerbach the error is here:
image

It looks like we might have to pin the dependency to the last sucessful version.

**Comment by networkfusion** [comment]
Actually, this is the error that terminates:
image

**Comment by networkfusion** [comment]
@Ellerbach 
image

**Comment by networkfusion** [comment]
I have made an autonimous PR here: https://github.com/nanoframework/nf-VSCodeExtension/pull/214
Which is building sucessfully as I also noticed the same problem with dependency updates.

**Comment by Ellerbach** [comment]
> I have made an autonomous PR here: #214 Which is building sucessfully as I also noticed the same problem with dependency updates.

I haven't seen before making the change myself. So seems like all green now

**Review by josesimoes — APPROVED** [review]
LGTM! Thanks for taking care of this.

**Pull request merged**